### PR TITLE
HTML: target=_blank implies noopener; opener support

### PR DESCRIPTION
**Pull request by annevk**
For https://github.com/whatwg/html/pull/4330.

### CODE REVIEW

**Comment by annevk** [comment]
@foolip I don't know why Chrome is failing here.

**Comment by foolip** [comment]
From the logs:

# All results ##

### /html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener.html ###
|                            Subtest                             |            Results            |                   Messages                   |
|----------------------------------------------------------------|-------------------------------|----------------------------------------------|
|                                                                | TIMEOUT                       |                                              |
| `Anchor element with target=_blank with rel=noopener`          | **PASS: 1/10, TIMEOUT: 9/10** | `Test timed out`                             |
| `Anchor element with target=_blank with rel=opener`            | PASS                          |                                              |
| `Anchor element with target=_blank with implicit rel=noopener` | FAIL                          | `assert_equals: expected false but got true` |
| `Anchor element with target=_blank with rel=opener+noopener`   | **PASS: 1/10, TIMEOUT: 9/10** | `Test timed out`                             |
| `Anchor element with target=_blank with rel=noopener+opener`   | TIMEOUT                       | `Test timed out`                             |
| `Area element with target=_blank with rel=noopener`            | TIMEOUT                       | `Test timed out`                             |
| `Area element with target=_blank with rel=opener`              | PASS                          |                                              |
| `Area element with target=_blank with implicit rel=noopener`   | FAIL                          | `assert_equals: expected false but got true` |
| `Area element with target=_blank with rel=opener+noopener`     | **PASS: 2/10, TIMEOUT: 8/10** | `Test timed out`                             |
| `Area element with target=_blank with rel=noopener+opener`     | **PASS: 1/10, TIMEOUT: 9/10** | `Test timed out`                             |

### /html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener_base.html ###
|                            Subtest                             |            Results            |                   Messages                   |
|----------------------------------------------------------------|-------------------------------|----------------------------------------------|
|                                                                | TIMEOUT                       |                                              |
| `Anchor element with target=_blank with rel=noopener`          | **PASS: 2/10, TIMEOUT: 8/10** | `Test timed out`                             |
| `Anchor element with target=_blank with rel=opener`            | PASS                          |                                              |
| `Anchor element with target=_blank with implicit rel=noopener` | FAIL                          | `assert_equals: expected false but got true` |
| `Anchor element with target=_blank with rel=opener+noopener`   | TIMEOUT                       | `Test timed out`                             |
| `Anchor element with target=_blank with rel=noopener+opener`   | TIMEOUT                       | `Test timed out`                             |
| `Area element with target=_blank with rel=noopener`            | TIMEOUT                       | `Test timed out`                             |
| `Area element with target=_blank with rel=opener`              | PASS                          |                                              |
| `Area element with target=_blank with implicit rel=noopener`   | FAIL                          | `assert_equals: expected false but got true` |
| `Area element with target=_blank with rel=opener+noopener`     | **PASS: 1/10, TIMEOUT: 9/10** | `Test timed out`                             |
| `Area element with target=_blank with rel=noopener+opener`     | **PASS: 3/10, TIMEOUT: 7/10** | `Test timed out`                             |

## Unstable results ##

|                                                  Test                                                 |                           Subtest                            |            Results            |     Messages     |
|-------------------------------------------------------------------------------------------------------|--------------------------------------------------------------|-------------------------------|------------------|
| `/html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener.html`      | `Anchor element with target=_blank with rel=noopener`        | **PASS: 1/10, TIMEOUT: 9/10** | `Test timed out` |
| `/html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener.html`      | `Anchor element with target=_blank with rel=opener+noopener` | **PASS: 1/10, TIMEOUT: 9/10** | `Test timed out` |
| `/html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener.html`      | `Area element with target=_blank with rel=opener+noopener`   | **PASS: 2/10, TIMEOUT: 8/10** | `Test timed out` |
| `/html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener.html`      | `Area element with target=_blank with rel=noopener+opener`   | **PASS: 1/10, TIMEOUT: 9/10** | `Test timed out` |
| `/html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener_base.html` | `Anchor element with target=_blank with rel=noopener`        | **PASS: 2/10, TIMEOUT: 8/10** | `Test timed out` |
| `/html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener_base.html` | `Area element with target=_blank with rel=opener+noopener`   | **PASS: 1/10, TIMEOUT: 9/10** | `Test timed out` |
| `/html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener_base.html` | `Area element with target=_blank with rel=noopener+opener`   | **PASS: 3/10, TIMEOUT: 7/10** | `Test timed out` |

What is the runtime of the test, might it be close to timing out?

**Comment by annevk** [comment]
It does. It shouldn't run any longer than the per-test timeout though.

**Comment by foolip** [comment]
I guess the one could only start trying to reproduce and minify until the cause is clear. Not sure who a good contact working on Chromium would be. @tkent-google, do you know who's are this is?

File: html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener_base.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  <script src=/resources/testharness.js></script>
+  <script src=/resources/testharnessreport.js></script>
+  <base target=_blank>
+</head>
+<body>
+  <a href="support/target_blank_iplicit_noopener.html?a1" id="a1" rel="noopener">Click me</a>
+  <a href="support/target_blank_iplicit_noopener.html?a2" id="a2" rel="opener">Click me</a>
+  <a href="support/target_blank_iplicit_noopener.html?a3" id="a3">Click me</a>
+  <a href="support/target_blank_iplicit_noopener.html?a4" id="a4" rel="opener noopener">Click me</a>
+  <a href="support/target_blank_iplicit_noopener.html?a5" id="a5" rel="noopener opener">Click me</a>
```
Comment (domenic) [inline html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener_base.html]: Test interaction with noreferrer too?

The use of base target=_blank seems really subtle.

Comment (annevk) [reply html/semantics/links/links-created-by-a-and-area-elements/target_blank_implicit_noopener_base.html]: Note that there's also a test without `<base>`. I figured we should test both.

**Comment by annevk** [comment]
Should be all good now, from a test perspective. Dunno about Taskcluster though.

**Comment by annevk** [comment]
This can now be merged. I'll be merging the HTML PR shortly. A Chromium bug is already filed as per that PR.

For squash and merge please only use the commit message from the initial commit.

**Comment by foolip** [comment]
@annevk squash merged!

**Comment by annevk** [comment]
Takk!

**Comment by ericlaw1979** [comment]
The test case does:

`
   <area />
</img>`

Is that valid?

**Comment by zcorpan** [comment]
`</img>` is invalid HTML. The association needs to be done with `<map name=foo>`

**Comment by annevk** [comment]
It's indeed bogus, but shouldn't affect the test. I'll create a PR to clean that up a bit though.

**Pull request merged**